pkg/sanity/controller.go

```diff
@@ -854,6 +854,43 @@ var _ = DescribeSanity("Controller Service", func(sc *SanityContext) {
 			cl.UnregisterVolume(name)
 		})
 
+		It("should fail when publishing volumes more than the node max attach limit", func() {
+
+			By("getting node id")
+			nodeInfo, err := n.NodeGetInfo(
+				context.Background(),
+				&csi.NodeGetInfoRequest{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(nodeInfo).NotTo(BeNil())
+
+			nid := nodeInfo.GetNodeId()
+			Expect(nid).NotTo(BeEmpty())
+
+			By("creating volumes")
+			name1 := uniqueString("sanity-max-attach-limit-vol1")
+			volID1, err := CreateAndControllerPublishVolume(sc, c, name1, nid)
+			Expect(err).NotTo(HaveOccurred())
+			cl.RegisterVolume(name1, VolumeInfo{VolumeID: volID1, NodeID: nid})
+
+			name2 := uniqueString("sanity-max-attach-limit-vol2")
+			volID2, err := CreateAndControllerPublishVolume(sc, c, name2, nid)
+			Expect(err).NotTo(HaveOccurred())
+			cl.RegisterVolume(name2, VolumeInfo{VolumeID: volID2, NodeID: nid})
+
+			name3 := uniqueString("sanity-max-attach-limit-vol3")
+			_, err = CreateAndControllerPublishVolume(sc, c, name3, nid)
+			Expect(err).To(HaveOccurred())
```

Comment:
This seems to assume that the driver has an attach limit of 2. This isn't going to be true for all drivers that are getting sanity tested.

The test needs to check the limit by querying the driver, skip the test if there is no limit, otherwise create n + 1 volumes to reach the limit.

Reply:
Right! I was thinking in terms of the mock driver only :sweat_smile:
Thanks a lot for pointing that out.

Reply:
Updated! Should be good now. :slightly_smiling_face:

Reply:
The world would be a simpler place if we only wrote software where test code only talks to other test code. But alas, there's also such a thing as production code :slightly_smiling_face:

Thanks for addressing this.